jsonrpc/src/test/connection.test.ts

```diff
@@ -497,4 +507,117 @@ suite('Connection', () => {
 		client.listen();
 		(client.sendNotification as Function)(type, 10);
 	});
+
+	test('Regular Cancellation', (done) => {
+		let type = new RequestType0<void, void>('cancelTest');
+		let duplexStream1 = new TestDuplex('ds1');
+		let duplexStream2 = new TestDuplex('ds2');
+
+		const source = new CancellationTokenSource();
+		let server = hostConnection.createMessageConnection(duplexStream2, duplexStream1, hostConnection.NullLogger);
+		server.onRequest(type, async t => {
+			source.cancel();
+
+			// wait until cancellation is set
+			while (!t.isCancellationRequested) {
+				await delay(0);
+			}
+
+			done();
+		});
+		server.listen();
+
+		let client = hostConnection.createMessageConnection(duplexStream1, duplexStream2, hostConnection.NullLogger);
+		client.listen();
+		client.sendRequest(type, source.token);
+	});
+
+	test('File Based Cancellation', (done) => {
```

Comment:
Shouldn't that test and the next one be in the new cancellation package

Reply:
if naming is issue, I can change it to CustomCancellationStrategy. I believe jsonrpc needs tests that confirm custom cancellation strategy works.